public/app/features/integrated-alerting/components/NotificationChannel/AddNotificationChannelModal/PagerDutyFields/PagerDutyFields.tsx

```diff
@@ -1,18 +1,38 @@
 import React, { FC } from 'react';
-import { TextInputField, validators } from '@percona/platform-core';
+import { RadioButtonGroupField, TextInputField, validators } from '@percona/platform-core';
 import { Messages } from '../AddNotificationChannelModal.messages';
 import { PagerDutyFieldsProps } from './PagerDutyFields.types';
+import { PagerDutyKeyType } from '../../NotificationChannel.types';
+import { PAGER_DUTY_TYPE_OPTIONS } from '../AddNotificationChannel.constants';
 
-export const PagerDutyFields: FC<PagerDutyFieldsProps> = ({ values }) => {
-  const { required } = validators;
-  const { routing, service } = values;
-  const routingValidators = service ? [] : [required];
-  const serviceValidators = routing ? [] : [required];
+const { required } = validators;
+const keyValidator = [required];
 
+export const PagerDutyFields: FC<PagerDutyFieldsProps> = ({ values, mutators }) => {
   return (
     <>
-      <TextInputField name="routing" label={Messages.fields.routingKey} validators={routingValidators} />
-      <TextInputField name="service" label={Messages.fields.serviceKey} validators={serviceValidators} />
+      <RadioButtonGroupField
+        inputProps={{
+          /**
+           * TODO We probably should change platform-core to allow to pass values to the options.
+           * That way, we can take the value from the event itself here and call the mutator, rather than checking the form values.
+           */
+          onInput: () => {
+            values?.keyType === PagerDutyKeyType.service
```

Comment:
Why do we need to reset the key here? Why not handle it on the submit function? That way we could remove mutators and simplify this code.

Reply:
I think the question is that the expected behaviour is that when you click one of the toggles, you should clear the other, so that if you go back to it, it'd be now reset. Is this right, @atymchuk ? If we're ok with letting both inputs stay there when the user clicks back and forth, then Tiago is right, if's way simple to just send the selected value when submitting instead of resetting values.

Reply:
For me as user I find not clearing the inputs a good thing, so if I go back I still have what I wrote. Anyway I don't think that's what's going to happen either way since we are removing the inputs from the DOM, so that's why I thought the reset was unnecessary.

Reply:
Yup, that's a good point. Let's hear what Alex has to say, but that'd make things a lot simpler for sure.
Removing / adding from the DOM is actually fine, as the value will come from the form's state (I only added `initialValues` for isolation purposes. Even if you remove it, the input gets populated with what comes from the outer form), that's why I had to force the cleaning via mutator.

Reply:
I'd keep the values the users entered.

Reply:
Got it!